pkg/controllers/provisioning/provisioner.go

```diff
@@ -446,6 +446,8 @@ func (p *Provisioner) getDaemonSetPods(ctx context.Context) ([]*v1.Pod, error) {
 func (p *Provisioner) Validate(ctx context.Context, pod *v1.Pod) error {
 	return multierr.Combine(
 		validateProvisionerNameCanExist(pod),
```

Comment:
Is this supported for the v1beta1 apis?

Reply:
That's a good point. I think we need to validate that both the labels have a `DoesNotExist` requirement; else, we are going to get into a situation where some pods _can_ actually schedule to Karpenter nodes that are managed by a NodePool.

I also need to add this detail to the Upgrade Guide. I'm not sure that there is a good way around this one.